types/next/index.d.ts

```diff
@@ -52,7 +52,7 @@ declare namespace next {
         /** Fetch Response object (client only) - from https://developer.mozilla.org/en-US/docs/Web/API/Response */
         jsonPageRes?: NodeResponse;
         /** Error object if any error is encountered during the rendering */
-        err?: Error;
+        err?: Error & {statusCode? : number};
```

Comment:
It looks like that the type of `err` is not `Error`. Guessing it's a `Response` object?

Reply:
Within the next code base they use [this `createError` function from `micro`](https://github.com/zeit/micro/blob/069f31fa506086cdbf0ce3364d16ab972e8cc200/lib/index.js#L19-L26):

```
const createError = (code, message, original) => {
	const err = new Error(message);

	err.statusCode = code;
	err.originalError = original;

	return err;
};
```

for the errors, appending this `statusCode`. They then [suggest in the next docs when creating your own `_error.js` page](https://github.com/zeit/next.js#custom-error-handling) to rely on this `statusCode` property.

Reply:
if we go with this approach I'd suggest including `originalError` for completeness as well